woocommerce/woocommerce-ios
# Fastfile: Add hotfix version number check

## Description
This PR replaces the use of the Release Toolkit's `ios_current_branch_is_hotfix` action with a method using the new versioning tooling. `ios_current_branch_is_hotfix` relies on the version xcconfig environment variable that we aren't using anymore. 

I also cleaned up a bit of the hotfix lane code to simplify it. 

I'm merging into the `release/16.8` branch because the `release_is_hotfix?` check is used during the `finalize_release` lane. 

## Testing instructions
CI will fail until https://github.com/woocommerce/woocommerce-ios/pull/11664 is merged into the `release/16.8` branch. But I'm confident these steps below are enough to verify the changes in this PR work as expected. Rubocop also passed locally with no violations. 

### To test the hotfix check
1. In the `finalize_release` lane, comment out everything after the first line, so that only ``UI.user_error!('To finalize a hotfix, please use the `finalize_hotfix_release` lane instead') if release_is_hotfix?`` remains
2. Run `bundle exec fastlane finalize_release`
3. The command will complete successfully because the version number in this branch is `16.8`
4. In `config/Version.Public.xcconfig`, change the `VERSION_SHORT` to `16.8.1` and save
5. Re-run `bundle exec fastlane finalize_release`
6. The command will show the error message ``UI.user_error!('To finalize a hotfix, please use the `finalize_hotfix_release` lane instead') if release_is_hotfix?``

### To test the hotfix code simplification
1. In the `new_hotfix_release` lane, comment out `ensure_git_status_clean` and everything after `UI.important(message)`
2. Run `bundle exec fastlane new_hotfix_release version:16.6.1`
3. The expected output is: 

```
      New hotfix version: 16.6.1
      New build code: 16.6.1.0
      Branching from tag: 16.6

```
